appservice/src/tree/DeploymentTreeItem.ts

```diff
@@ -156,6 +156,15 @@ export class DeploymentTreeItem extends AzureTreeItem<ISiteTreeRoot> {
         }
     }
 
+    private getDeploymentMessage(message: string): string {
```

Comment:
Now that you created a `getDeploymentMessage` method, I'd prefer to move all message-related logic into this method.

Something like this:
```typescript
private getDeploymentMessage(deployResult: DeployResult): string {
    let message: string = nonNullProp(deployResult, 'message');
    try {
        const messageJSON: { message?: string } = <{ message?: string }>JSON.parse(message);
        if (messageJSON.message) {
            message = messageJSON.message;
        }
    } catch {
        // ignore and assume message was not in JSON format
    }

    const firstLine: string = this.getFirstLine(message);
    // truncate long messages and add "..."
    return firstLine.length > 50 ? `${firstLine.substring(0, 50)}...` : firstLine;
}
```
(I didn't test this, but it was pretty simple copy-paste. Please double check what I did)

Reply:
Sure, will update it.

Reply:
Done. I have updated as suggested. Thanks for your review !